core/src/main/java/co/cask/wrangler/clients/SchemaRegistryClient.java

```diff
@@ -0,0 +1,289 @@
+  }
+
+  /**
+   * Returns a instance of {@link SchemaRegistryClient} initialized with service url.
+   *
+   * @param context of the pipeline in which this client is invoked.
+   * @return instance of this class.
+   */
+  public static SchemaRegistryClient getInstance(PipelineContext context) {
+    return new SchemaRegistryClient(context.getService("dataprep", "service").toString());
+  }
+
+  /**
+   * Retrieves schema provided a schema id and version of the schema.
+   *
+   * @param id of the schema.
+   * @param version of the schema.
+   * @return {@link Response} of the schema.
+   * @throws URISyntaxException thrown if there are issue with construction of url.
+   * @throws IOException throw when there are issues connecting to the service.
+   * @throws RestClientException thrown when there are issues with request or response returned.
+   */
+  public byte[] getSchema(String id, long version)
+    throws URISyntaxException, IOException, RestClientException {
+    URL url = concat(new URI(baseUrl), String.format("schemas/%s/versions/%d", id, version)).toURL();
+    Response<SchemaInfo> response = request(url, "GET", new TypeToken<Response<SchemaInfo>>(){}.getType());
+    if (response.getCount() == 1) {
+      return Bytes.fromHexString(response.getValues().get(0).getSpecification());
+    }
+    return null;
+  }
+
+  /**
+   * Retrieves schema provided a schema id. It provides the latest, current version of schema.
+   *
+   * @param id of the schema.
+   * @return {@link SchemaInfo} of the schema if ok, else null.
+   * @throws URISyntaxException thrown if there are issue with construction of url.
+   * @throws IOException throw when there are issues connecting to the service.
+   * @throws RestClientException thrown when there are issues with request or response returned.
+   */
+  public byte[] getSchema(String id)
+    throws URISyntaxException, IOException, RestClientException {
+    URL url = concat(new URI(baseUrl), String.format("schemas/%s", id)).toURL();
+    Response<SchemaInfo> response = request(url, "GET", new TypeToken<Response<SchemaInfo>>(){}.getType());
+    if (response.getCount() == 1) {
+      return Bytes.fromHexString(response.getValues().get(0).getSpecification());
+    }
+    return null;
+  }
+
+  /**
+   * Gets all the versions of schemas given a schema id.
+   *
+   * @param id of the schema.
+   * @return a list of schema versions.
+   * @throws URISyntaxException thrown if there are issue with construction of url.
+   * @throws IOException throw when there are issues connecting to the service.
+   * @throws RestClientException thrown when there are issues with request or response returned.
+   */
+  public List<Long> getVersions(String id)
+    throws URISyntaxException, IOException, RestClientException {
+    URL url = concat(new URI(baseUrl), String.format("schemas/%s/versions", id)).toURL();
+    Response<Long> response = request(url, "GET", new TypeToken<Response<Long>>(){}.getType());
```

Comment:
it is technically possible for request() to return null. we should check and return empty list.